packages/block-editor/src/components/block-popover/index.js

```diff
@@ -40,15 +40,30 @@ export default function BlockPopover( {
 		};
 	}, [ selectedElement, lastSelectedElement, __unstableRefreshSize ] );
 
-	if ( ! selectedElement || ( bottomClientId && ! lastSelectedElement ) ) {
-		return null;
-	}
-
 	const anchorRef = {
 		top: selectedElement,
 		bottom: lastSelectedElement,
 	};
 
+	const anchorOwnerDocument = anchorRef.top.ownerDocument;
+
+	const popoverOffset = useMemo( () => {
+		const frameElement = anchorOwnerDocument?.defaultView?.frameElement;
+
+		if ( ! frameElement || anchorOwnerDocument === document ) {
+			return;
+		}
+
+		const { left, top } = frameElement.getBoundingClientRect();
+		return {
+			mainAxis: -top,
+			crossAxis: -left,
+		};
+	}, [ anchorOwnerDocument ] );
```

Comment:
This was mostly an experiment to see what it would take to remove the "iframe-related" logic from the `Popover` component — but given that we can't really remove the scroll listener at the moment, we can probably keep the whole thing for now.

I've added the internal iframe middleware in https://github.com/WordPress/gutenberg/pull/42531/commits/e38e1b72b225eb5e6756cf3f05c2b3e6fbb63280, but I've refactored it to use the `offset` middleware instead. In fact, without using the `offset` middleware, the popover was being shifted way too far up when scrolling the achor below the fold.

I've also taken the opportunity to factor in the `placement` for the `limitShift` utility in https://github.com/WordPress/gutenberg/pull/42531/commits/97124da46bf8ebd389fa1ebd78e2df22759d7441